### PR TITLE
fix: theme type check fix

### PR DESCRIPTION
**Pull request by hoob3rt**
Handles weird theme types in config & makes code cleaner imo

### CODE REVIEW

File: lua/lualine/init.lua [hunk 1]
```diff
@@ -266,23 +266,21 @@ end
...
+      if ok then return theme end
+    elseif type(theme_name) == 'table' then
+      -- use the provided theme as-is
+      return config.options.theme
     end
-  else
-    -- use the provided theme as-is, assuming it's a dictionary
-    theme = config.options.theme
+    vim.api.nvim_echo({
+      {'theme ' .. theme_name .. ' not found defaulting to gruvbox', 'ErrorMsg'}
```
Comment (shadmansaleh) [inline lua/lualine/init.lua:279]: string concat will fail for invalid types like boolean

Comment (hoob3rt) [reply lua/lualine/init.lua]: Yeah you are right, should be resolved now

File: lua/lualine/init.lua [hunk 2]
```diff
@@ -266,23 +266,24 @@ end
...
-    theme = config.options.theme
+    vim.api.nvim_echo({
+      {
+        'theme ' .. tostring(theme_name) .. ' not found, defaulting to gruvbox',
+        'ErrorMsg'
+      }
+    }, true, {})
+    return require 'lualine.themes.gruvbox'
   end
+  local theme = get_theme_from_config();
```
Comment (shadmansaleh) [inline lua/lualine/init.lua:286]: Why is the semicolon here though? Otherwise LGTM .

**Comment by shadmansaleh** [comment]
@hoob3rt can you give me some access to the repo to merge bugfixes ?

File: lua/lualine/init.lua (hunk 2 above)
Comment (hoob3rt) [reply lua/lualine/init.lua:286]: oof, c and ts habit

**Comment by hoob3rt** [comment]
> @hoob3rt can you give me some access to the repo to merge bugfixes ?

yeah I will look into it in ~4 hours at the latest. I have classes & meetings rn

**Comment by hoob3rt** [comment]
> > @hoob3rt can you give me some access to the repo to merge bugfixes ?
> 
> yeah I will look into it in ~4 hours at the latest. I have classes & meetings rn

Okay I've actually done it now, there is no option that I'm aware of to give you a `moderator` type permission. So I would like to ask you to merge only bug fixes & obviously never commit to master

**Pull request merged**